From 2efc79702a47f9cbbeecbb2f604577f1ec8c95ba Mon Sep 17 00:00:00 2001
From: Yogesh Siraswar <yogeshs@ti.com>
Date: Wed, 8 Dec 2021 17:50:19 -0600
Subject: [KIG PATCH] Makefile: Skip signing of binaries for combined boot hs
 images

Combined sysfw binaries used in tiboot.bin do not require signed images.
This fixes issues with AM64x HS and J7200 HS boot images

Upstream-Status: Submitted
Signed-off-by: Yogesh Siraswar <yogeshs@ti.com>
---
 Makefile           | 8 +++++++-
 soc/am64x/Makefile | 1 +
 soc/j7200/Makefile | 1 +
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 47b3b20..ab465df 100644
--- a/Makefile
+++ b/Makefile
@@ -215,8 +215,14 @@ $(soc_objroot)/%.o: %.c
 	$(CROSS_COMPILE)gcc $(CFLAGS) -c -o $@-pre-validated $<
 	python3 ./scripts/sysfw_boardcfg_validator.py -b $@-pre-validated -i -o $@ -s $(SOC) -l $@.log
 
-# On HS board configuration binaries must be signed
+# On HS board configuration binaries must be signed unless it is combined boot image
 ifdef HS
+ifndef COMBINED_BOOT_IMAGE
+SIGN_BINARY=1
+endif
+endif
+
+ifdef SIGN_BINARY
 %.bin.unsigned: %.o
 	$(CROSS_COMPILE)objcopy -S -O binary $< $@
 %.bin: %.bin.unsigned
diff --git a/soc/am64x/Makefile b/soc/am64x/Makefile
index bfbe3e5..33ebbd0 100644
--- a/soc/am64x/Makefile
+++ b/soc/am64x/Makefile
@@ -31,6 +31,7 @@
 #
 
 SBL_LOADADDDR ?= 0x70000000
+COMBINED_BOOT_IMAGE = 1
 COMBINED_SYSFW_BRDCFG_LOADADDR ?= 0x7b000
 LOADADDR  ?= 0x44000
 SCIFS = sci
diff --git a/soc/j7200/Makefile b/soc/j7200/Makefile
index 3b754f7..0baef51 100644
--- a/soc/j7200/Makefile
+++ b/soc/j7200/Makefile
@@ -31,6 +31,7 @@
 #
 
 SBL_LOADADDDR ?= 0x41c00000
+COMBINED_BOOT_IMAGE = 1
 COMBINED_TIFS_BRDCFG_LOADADDR ?= 0x7f000
 COMBINED_DM_BRDCFG_LOADADDR ?= 0x41c80000
 LOADADDR ?= 0x40000
-- 
2.17.1

